cpython/Makefile

```diff
@@ -30,6 +30,12 @@ BZIP2TARBALL=$(ROOT)/downloads/bzip2-1.0.2.tar.gz
 BZIP2BUILD=$(ROOT)/build/bzip2-1.0.2
 BZIP2URL=ftp://sources.redhat.com/pub/bzip2/v102/bzip2-1.0.2.tar.gz
 
+# Add Mac system libraries (such as zlib) to the include path.
+# Required sometimes on OS X 10.14, always on OS X 10.15+.
+ifeq ($(shell uname),Darwin)
+	MAC_SDK_PATH := $(shell xcrun --show-sdk-path)
+	MAC_BUILD_VARS := CFLAGS=-I$(MAC_SDK_PATH)/usr/include CXXFLAGS=-I$(MAC_SDK_PATH)/usr/include
+endif
```

Comment:
I suspect this isn't tested by CircleCI...  Does anyone have another Catalina (or pre-Catalina) Mac that they could test it with?

Reply:
I can take a look, I have both Catalina + a pre Catalina macs kicking around

Reply:
We explicitly *don't* want the system libraries to be found here, since emscripten provides specific versions of them (which are probably different from the ones provided by the system).  Do you have a specific error that this was intended to solve?  Maybe there's another way...

Reply:
I think this is trying to workaround the fact that it is no longer possible to install system headers in /usr/include on Catalina, due to system integrity protection: 

https://apple.stackexchange.com/questions/372032/usr-include-missing-on-macos-catalina-with-xcode-11

IIRC the hostpython config.log is used later for building the webassembly version of pyodide?  If so we may need to figure out how to temporarily use the headers from the SDK for bootstrapping purposes.

I suspect this may have something to do with the kiwisolver issue I mention below (https://github.com/iodide-project/pyodide/pull/590#pullrequestreview-337810712)

Reply:
The specific issue that I'm trying to solve is that Python decides to not build with zlib support, which subsequently breaks pip.

I see that pyodide does download zlib, but I didn't immediately see where it's built for the host?

Reply:
I see.  This is probably ok for that purpose (and these config are only added to "native" builds of Python, not the WebAssembly one) -- so I retract my comment.  This seems fine if it works.

Reply:
@mdboom are you sure this config is only used for native builds of python? it seems like somehow kiwisolver is trying to build with these flags later on (see my comment below).

Reply:
Oh right -- they are used in the native Python build, but the way Python works, it takes those flags and generates `sysconfig` information, which it later uses those flags to build extensions.  So if this is needed here to build the native Python, we would have to edit the files it generates to remove them again for when it builds extensions for WebAssembly.  Or maybe we just need to provide a hard-coded sysconfig for the WebAssembly environment that doesn't include any system paths.  All that's kind of messy.

But seemingly Python itself builds on Catalina without requiring explicit workaround like this.  It looks like the Python [CHANGELOG](https://www.python.org/downloads/release/python-375/) includes at least some specific changes for Catalina compatibility.  Maybe upgrading to the latest Python in the 3.7 series will help and not require this workaround.

Reply:
Ah yes, it looks like https://github.com/python/cpython/commit/c7302116573d853d3181133477d9d0e4d4d3abfd might resolve our issues here. Definitely can't hurt to be using the latest minor version of 3.7 in any case

Reply:
Upgrading to 3.7.6 seems to fix this issue, and I can now finish compilation on MacOS X without the changes above. I filed #594 for that.